plugins/action/network.py

```diff
@@ -277,22 +277,42 @@ def _patch_update_module(self, module, task_vars):
         """
         import copy
 
-        from ansible.module_utils.basic import AnsibleModule as _AnsibleModule
+        from ansible_collections.ansible.netcommon.plugins.module_utils.network.common.utils import (
+            PatchedAnsibleModule,
+        )
 
-        # build an AnsibleModule that doesn't load params
-        class PatchedAnsibleModule(_AnsibleModule):
+        # update the task args w/ all the magic vars
+        self._update_module_args(self._task.action, self._task.args, task_vars)
+
+        # Create a class that inherits from our consolidated PatchedAnsibleModule
+        # but doesn't load params and can record results
+        class NetworkPatchedAnsibleModule(PatchedAnsibleModule):
```

Comment:
What is the purpose of overriding `PatchedAnsibleModule` here? This class overrides both methods on that class, and doesn't make use of the custom `__init__` logic.